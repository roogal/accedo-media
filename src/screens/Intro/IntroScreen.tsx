/*
 * @author BSG <dev@bsgroup.eu>
 * @copyright Better Software Group S.A.
 * @version: 1.0
 */
import React from "react";
import {
  Link
} from "react-router-dom";

import { ROUTES } from "../..//constants";

import styles from "./IntroScreen.module.scss";

export const IntroScreen = () => {
  return (
    <div className={styles.Container}>
      <div className={styles.Gradient}></div>
      <img className={styles.Logo} src="/logo192.png" alt="Accedo media" />
      <div className={styles.Content}>
        <div className={styles.Info}>
          <h1>Unlimited movies, TV shows, and more.</h1>
          <h2>Watch anywhere. Cancel anytime.</h2>
          <h3 className={styles.H3}>Ready to watch?</h3>
          <Link to={ROUTES.HOME}>
            <button className={styles.Button}>Get started</button>
          </Link>
        </div>
      </div>
    </div>
  );
};
