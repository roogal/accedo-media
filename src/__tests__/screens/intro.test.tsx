import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { IntroScreen } from "../../screens/Intro/IntroScreen";

describe("Intro", () => {
  it("renders a heading", () => {
    render(
      <BrowserRouter>
        <IntroScreen />
      </BrowserRouter>
    );

    const heading = screen.getByRole("heading", {
      name: "Unlimited movies, TV shows, and more.",
    });

    expect(heading).toBeInTheDocument();
  });

  it("renders a get started button", () => {
    render(
      <BrowserRouter>
        <IntroScreen />
      </BrowserRouter>
    );

    const button = screen.getByRole("button");

    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent("Get started");
  });

  it("renders intro unchanged", () => {
    const { container } = render(
      <BrowserRouter>
        <IntroScreen />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
