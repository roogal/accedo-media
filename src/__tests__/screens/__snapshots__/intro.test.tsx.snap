// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro renders intro unchanged 1`] = `
<div>
  <div
    class="Container"
  >
    <div
      class="Gradient"
    />
    <img
      alt="Accedo media"
      class="Logo"
      src="/logo192.png"
    />
    <div
      class="Content"
    >
      <div
        class="Info"
      >
        <h1>
          Unlimited movies, TV shows, and more.
        </h1>
        <h2>
          Watch anywhere. Cancel anytime.
        </h2>
        <h3
          class="H3"
        >
          Ready to watch?
        </h3>
        <a
          href="/home"
        >
          <button
            class="Button"
          >
            Get started
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
